synapse/storage/databases/main/events.py

```diff
@@ -1619,9 +1619,15 @@ def prefill():
 
         txn.call_after(prefill)
 
-    def _store_redaction(self, txn, event):
-        # invalidate the cache for the redacted event
+    def _store_redaction(self, txn: LoggingTransaction, event: EventBase) -> None:
+        # invalidate the caches for the redacted event
         txn.call_after(self.store._invalidate_get_event_cache, event.redacts)
+        self.store._invalidate_cache_and_stream(
+            txn, self.store.get_relations_for_event, (event.redacts,)
+        )
+        self.store._invalidate_cache_and_stream(
+            txn, self.store.get_applicable_edit, (event.redacts,)
+        )
```

Comment:
I don't think we need to stream the cache invalidation here, as they happen as part of the event replication? Removing it means we can shove less down the replication streams

Reply:
Ah, you're right. The other bit of code I pointed to will get called after replication!